src/main/telemetry/srxl.c

```diff
@@ -160,6 +162,168 @@ bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
     return true;
 }
 
+#if defined(USE_GPS) 
+
+// From Frsky implementation
+static void GPStoDDDMM_MMMM(int32_t mwiigps, gpsCoordinateDDDMMmmmm_t *result)
+{
+    int32_t absgps, deg, min;
+    absgps = ABS(mwiigps);
+    deg    = absgps / GPS_DEGREES_DIVIDER;
+    absgps = (absgps - deg * GPS_DEGREES_DIVIDER) * 60;        // absgps = Minutes left * 10^7
+    min    = absgps / GPS_DEGREES_DIVIDER;                     // minutes left
+    result->dddmm = deg * 100 + min;
+    result->mmmm  = (absgps - min * GPS_DEGREES_DIVIDER) / 1000;
+}
+
+uint32_t dec2bcd_r(uint16_t dec)
+{
+    return (dec) ? ((dec2bcd_r( dec / 10 ) << 4) + (dec % 10)) : 0;
```

Comment:
This is a recursive function (from stackoverflow?), and I'd advise against this and rewrite it into a loop.

Reply:
Yes, you are absolutely right. It was working, it was short so I used it. Unfortunately I don't know how to measure the real performance of both approaches on ARM, but I have written a simple iterative equivalent:

```
static uint32_t dec2bcd(uint16_t dec)
{
    uint32_t result = 0;
    uint8_t counter = 0;

    while (dec % 10 > 0) {
        result |= (dec % 10) << counter * 4;
        counter++;
        dec /= 10;
    }
    return result;
}
```

I also made it static. Does it look like a satisfying solution ?

Reply:
I've written a little python script for you that implements the code above:
```python
import numpy as np

def dec2bcd(dec):
    dec = np.uint16(dec)
    result = np.uint32(0)
    counter = np.uint8(0)
    while ((dec % 10) > 0):
        result |= (dec % 10) << counter * 4
        counter += 1
        dec = np.uint16(dec / 10)
    return np.uint32(result)

test_dec = np.uint16(0)

while (test_dec <= 30):
    test_bcd = dec2bcd(test_dec)
    print("dec2bcd({:d}) = {:d}".format(test_dec, test_bcd))
    test_dec += 1
```

Looking at the output one can notice that this doesn't work for multiples of 10:
```
dec2bcd(0) = 0
dec2bcd(1) = 1
dec2bcd(2) = 2
dec2bcd(3) = 3
dec2bcd(4) = 4
dec2bcd(5) = 5
dec2bcd(6) = 6
dec2bcd(7) = 7
dec2bcd(8) = 8
dec2bcd(9) = 9
dec2bcd(10) = 0
dec2bcd(11) = 17
dec2bcd(12) = 18
dec2bcd(13) = 19
dec2bcd(14) = 20
dec2bcd(15) = 21
dec2bcd(16) = 22
dec2bcd(17) = 23
dec2bcd(18) = 24
dec2bcd(19) = 25
dec2bcd(20) = 0
```

Reply:
You need a do { } while instead like this:
```python
import numpy as np

def dec2bcd(dec):
    dec = np.uint16(dec)
    result = np.uint32(0)
    counter = np.uint8(0)
    while True:
        result |= (dec % 10) << counter * 4
        counter += 1
        dec = np.uint16(dec / 10)
        if not ((dec % 10) > 0):
            break
    return np.uint32(result)

test_dec = np.uint16(0)

while (test_dec <= 30):
    test_bcd = dec2bcd(test_dec)
    print("dec2bcd({:d}) = {:d}".format(test_dec, test_bcd))
    test_dec += 1
```

Reply:
Shame on me. That's an obvious mistake. I should be beaten with a stick for not testing this carefully. Thanks for pointing this out. I think it should be enough to just replace `while (dec % 10 > 0)` with `while (dec)`. As I don't have access to any other programming stuff at the moment I have tested it as a Java method and it passes tests for all integers between 0 and 65535. Will check again with C version later today.

```java
    private static int dec2bcd (int dec) {
        int counter = 0;
        int result  = 0;
        
        while (dec > 0) {
            result |= (dec % 10) << counter * 4;
            counter++;
            dec /= 10;
        }
        return result;
    }

```